### PR TITLE
[keyboard] ymdk/id75

### PR DESCRIPTION
**Pull request by lesshonor**
## Description

YMDK is selling a new APM32-powered "ID75". Source code has been made available to several individuals upon request. I have tried to bring this code in line with current QMK guidelines so it can be upstreamed for improved user-friendliness.

Many thanks to @docteurzoidberg for testing my changes, and to zvecr for bringing in uf2boot.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/id75/chconf.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* Copyright 2020 QMK
```
Comment (zvecr) [inline keyboards/ymdk/id75/chconf.h:1]: This file should be removed

File: keyboards/ymdk/id75/info.json [hunk 2]
```diff
@@ -0,0 +1,208 @@
+{
+    "manufacturer": "YMDK",
+    "keyboard_name": "Idobao x YMDK ID75",
+    "maintainer": "qmk",
+    "bootloader": "uf2boot",
+    "board": "STM32_F103_STM32DUINO",
```
Comment (zvecr) [inline keyboards/ymdk/id75/info.json:6]: This board file isnt set up for `uf2boot`, so wouldnt work?
```suggestion
```

File: keyboards/ymdk/id75/info.json [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            { "flags": 2, "x": 224, "y": 0 }
+        ],
+        "max_brightness": 128
+    },
+    "rgblight": {
+        "pin": "B9"
+    },
+    "tapping": {
+        "toggle": 2
+    },
```
Comment (zvecr) [inline keyboards/ymdk/id75/info.json:120]:
```suggestion
```

File: keyboards/ymdk/id75/keymaps/via/keymap.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS"
+        ],
+        [
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",
+            "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS"
+        ]
+    ]
```
Comment (zvecr) [inline keyboards/ymdk/id75/keymaps/via/keymap.json:37]:
```suggestion
    ],
    "config": {
        "features": {
            "via": true,
            "lto": true
        }
    }
```

File: keyboards/ymdk/id75/keymaps/via/rules.mk [hunk 5]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/ymdk/id75/keymaps/via/rules.mk:1]: This file can be removed if the keymap.json suggestion is accepted

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/id75/info.json (hunk 2 above)
Comment (zvecr) [reply keyboards/ymdk/id75/info.json:6]: Ignore me for now then

File: keyboards/ymdk/id75/chconf.h (hunk 1 above)
Comment (docteurzoidberg) [reply keyboards/ymdk/id75/chconf.h:1]: compiled and flashed without: no noticable difference, i confirm

File: keyboards/ymdk/id75/info.json [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+            { "flags": 2, "x": 114, "y": 0 },
+            { "flags": 2, "x": 152, "y": 0 },
+            { "flags": 2, "x": 190, "y": 0 },
+            { "flags": 2, "x": 224, "y": 0 }
+        ],
+        "max_brightness": 128
+    },
+    "rgblight": {
+        "pin": "B9"
+    },
```
Comment (waffle87) [inline keyboards/ymdk/id75/info.json:117]:
```suggestion
```

File: keyboards/ymdk/id75/config.h [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#if defined(RGB_MATRIX_ENABLE)
+
```
Comment (waffle87) [inline keyboards/ymdk/id75/config.h:20]:
```suggestion
#    define RGB_DI_PIN B9
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/ymdk/id75/readme.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Idobao x YMDK ID75
+
+![Idobao x YMDK ID75](https://imgur.com/bhVfzrzh.jpg)
```
Comment (drashna) [inline keyboards/ymdk/id75/readme.md:3]:
```suggestion
![Idobao x YMDK ID75](https://i.imgur.com/bhVfzrzh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**